pkg/gardenlet/controller/backupentry/reconciler.go

```diff
@@ -356,6 +352,8 @@ func (r *Reconciler) deleteBackupEntry(
 			}
 			log.Info("Extension BackupEntry not yet deleted", "extensionBackupEntry", client.ObjectKeyFromObject(extensionBackupEntry))
 			return reconcile.Result{RequeueAfter: RequeueDurationWhenResourceDeletionStillPresent}, nil
+		} else if !apierrors.IsNotFound(err) {
+			return reconcile.Result{}, err
```

Comment:
What is the purpose of this change? 👀

Reply:
The linter was issuing a warning about the old check because in the `else` condition `err` is actually always `nil`
```
if err := r.SeedClient.Get(seedCtx, client.ObjectKeyFromObject(extensionBackupEntry), extensionBackupEntry); err != nil {
			if !apierrors.IsNotFound(err) {
				return reconcile.Result{}, err
			}
		} else if err == nil {
```

However, if I just leave it with `} else {` then the linter issues an error this time that the code could be optimised further - and that's the change I've done here.

Reply:
Which linter? 👀 I don't see any failing runs of the unit test PR check

Reply:
I should have been more specific. The initial finding was from https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/nilness#cond

However, if you just remove the nil check and leave it like:
```
if err := r.SeedClient.Get(seedCtx, client.ObjectKeyFromObject(extensionBackupEntry), extensionBackupEntry); err != nil {
...
} else {
...
}
```
`golangci-lint` throws an error: 
```
Executing golangci-lint
pkg/gardenlet/controller/backupentry/reconciler.go:344:3: early-return: if c { ... } else { ... return } can be simplified to if !c { ... return } ... (move short variable declaration to its own line if necessary) (revive)
                if err := r.SeedClient.Get(seedCtx, client.ObjectKeyFromObject(extensionBackupEntry), extensionBackupEntry); err != nil {
                        if !apierrors.IsNotFound(err) {
                                return reconcile.Result{}, err
                        }
                } else {
                        if lastError := extensionBackupEntry.Status.LastError; lastError != nil {
                                r.Recorder.Event(backupEntry, corev1.EventTypeWarning, gardencorev1beta1.EventDeleteError, lastError.Description)

                                if updateErr := r.updateBackupEntryStatusError(gardenCtx, backupEntry, operationType, lastError.Description, lastError); updateErr != nil {
                                        return reconcile.Result{}, fmt.Errorf("could not update status after deletion error: %w", updateErr)
                                }
                                return reconcile.Result{}, errors.New(lastError.Description)
                        }
                        log.Info("Extension BackupEntry not yet deleted", "extensionBackupEntry", client.ObjectKeyFromObject(extensionBackupEntry))
                        return reconcile.Result{RequeueAfter: RequeueDurationWhenResourceDeletionStillPresent}, nil
                }
```

Reply:
This is from the go language server `gopls`. You're right that it wouldn't be used by everyone - and I'm not sure if we can enforce it. 
I added the change to the PR because I noticed it while working on the PR and thought the deletion logic would look a bit cleaner. But I can remove it and add it later on if we think it's worth it.